src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+
+/**
+ * Handles the loading and saving of data onto the local storage by serialising the TaskList object
+ */
+public class Storage {
+    // hardcoded for now:
+    public static final Path ROOT_PATH = Paths.get("").toAbsolutePath();
+    public static final Path DATA_DIR_PATH = Paths.get(ROOT_PATH.toAbsolutePath().toString(), "data");
+    public static final Path SAVE_FILE_PATH = Paths.get(DATA_DIR_PATH.toAbsolutePath().toString(), "savedData");
+    /**
+     * Saves the current TaskList object by serialising it and writing onto the default file
+     *
+     * @param allTasks Current TaskList that represents all the tasks that have been created by the user
+     *
+     * @throws IOException If there are errors writing onto the intended file
+     */
+    public static void save(TaskList allTasks) throws IOException {
+        initStorage();
+        try {
+            FileOutputStream fos = new FileOutputStream(SAVE_FILE_PATH.toFile());
```

Comment:
I am not sure about this but I want to point out that the variable names generated from abbreviation may be confusing in some ways. For example, "fos" makes me think of faculty of science...  I think it is acceptable to write in  this way, but I believe use complete words can reduce confusion (such as fileOutputStream)

Reply:
hmm, the scope for the `fos` and `oos` is limited to the try block only. I doubt it introduces any ambiguity since these var names are mere abbreviations of their respective classes.